api/client/registry.go

```diff
@@ -91,6 +91,10 @@ func (cli *DockerCli) ConfigureAuth(flUser, flPassword, serverAddress string, is
 		cli.in = os.Stdin
 	}
 
+	if !isDefaultRegistry {
```

Comment:
The `isDefaultRegistry` flag is a bit weird and I'm not sure whether it can be trusted here.

I see that `RegistryAuthenticationPrivilegedFunc` has the following code:

```go
                fmt.Fprintf(cli.out, "\nPlease login prior to %s:\n", cmdName)
                indexServer := registry.GetAuthConfigKey(index)
                authConfig, err := cli.ConfigureAuth("", "", indexServer, false)
```

It hardcodes `isDefaultRegistry = false`, but `GetAuthConfig` may return the default registry, including the scheme:

```go
// GetAuthConfigKey special-cases using the full index address of the official
// index as the AuthConfig key, and uses the (host)name[:port] for private indexes.
func GetAuthConfigKey(index *registrytypes.IndexInfo) string {
        if index.Official {
                return IndexServer
        }
        return index.Name
}
```

`IndexServer` includes the scheme.

So I think that this code path might wrongly strip the scheme from the default registry. Would it make sense to remove the `isDefaultRegistry` flag (which doesn't seem to be set correctly), and instead have `ConfigureAuth` check if `serverAddress == ElectAuthServer(ctx)` instead?